jakarta-eclipse/src/org/jakarta/codeActionProposal/ChangeCorrectionProposal.java

```diff
@@ -0,0 +1,92 @@
+package org.jakarta.codeActionProposal;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.manipulation.ChangeCorrectionProposalCore;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.IUndoManager;
+import org.eclipse.ltk.core.refactoring.RefactoringCore;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+
+
+public class ChangeCorrectionProposal extends ChangeCorrectionProposalCore {
+	// LSP: Code Action Kind
+	private String fKind;
+
+	/**
+	 * Constructs a change correction proposal.
+	 *
+	 * @param name      the name that is displayed in the proposal selection dialog
+	 * @param change    the change that is executed when the proposal is applied or
+	 *                  <code>null</code> if the change will be created by
+	 *                  implementors of {@link #createChange()}
+	 */
+	public ChangeCorrectionProposal(String name, String kind, Change change) {
+		super(name, change, 0);
+		fKind = kind;
+	}
+
+	/**
+	 * Performs the change associated with this proposal.
+	 * <p>
+	 * Subclasses may extend, but must call the super implementation.
+	 *
+	 * @throws CoreException when the invocation of the change failed
+	 */
+	@Override
+	protected void performChange() throws CoreException {
+
+		Change change = null;
+		try {
+			change = getChange();
+			if (change != null) {
+
+				change.initializeValidationData(new NullProgressMonitor());
+				RefactoringStatus valid = change.isValid(new NullProgressMonitor());
+				if (valid.hasFatalError()) {
+					IStatus status = new Status(IStatus.ERROR, MicroProfileCorePlugin.PLUGIN_ID, IStatus.ERROR,
```

Comment:
in the example code, they have a pluginID being returned when there is error occurred. Is there anything I should return here?

Reply:
You can use the plugin id we set here: https://github.com/MicroShed/jakarta-ls/blob/master/jakarta-eclipse/src/org/jakarta/lsp4e/Activator.java#L14

Reply:
Great, I'll do that! thank you